src/main/java/kostka/moviecatalog/service/MovieDetailService.java

```diff
@@ -78,10 +85,41 @@ private MovieDetail populateMovieDetail(final Movie movie,
         movieDetail.setDirector(movie.getDirector());
         movieDetail.setDescription(movie.getDescription());
         movieDetail.setAverageRating(movie.getAverageRating());
-        movieDetail.setComments(comments);
-        movieDetail.setRatings(ratings);
+        movieDetail.setComments(getCommentDtoListForMovieDetail(comments));
+        movieDetail.setRatings(getRatingDtoListForMovieDetail(ratings));
         movieDetail.setBought(isBoughtByUser);
         movieDetail.setForAdults(movie.isForAdults());
         return movieDetail;
     }
+
+    private List<CommentDetailDto> getCommentDtoListForMovieDetail(final List<Comment> comments) {
+         return comments.stream()
+                .map(this::getCommentDtoForMovieDetail)
+                .collect(Collectors.toList());
+    }
+
+    private CommentDetailDto getCommentDtoForMovieDetail(final Comment comment) {
+        User user = userService.getUser(comment.getAuthorId());
+        return new CommentDetailDto(comment.getCommentId(), getAuthorName(user), comment.getCommentText());
+    }
+
+    private List<RatingDetailDto> getRatingDtoListForMovieDetail(final List<Rating> ratings) {
+        return ratings.stream()
+                .map(this::getRatingDtoForMovieDetail)
+                .collect(Collectors.toList());
+    }
+
+    private RatingDetailDto getRatingDtoForMovieDetail(final Rating rating) {
+        User user = userService.getUser(rating.getAuthorId());
+        return new RatingDetailDto(rating.getRatingId(), rating.getRatingValue(), getAuthorName(user));
+    }
+
+    private String getAuthorName(final User user) {
```

Comment:
nit: tuhle metodu bych možná dal klidně přímo do `User`.

Reply:
to se mi líbí, díky ;)

Reply:
Jinak, když už děláš ty patterny... ta metoda by mohlá být static a pak se jí někdy říká Extension Method.

Metoda, která chováním dává smysl mít u jiné třídy. Ale nemůžeš ji tam přidat, protože k ní nemáš zdroják, nebo je final atd. Tak ji dáš do jiné třídy a jako parametr bere ten objekt, na kterém by ta metoda normálně byla.

Reply:
kdybych ji nechal tak jak byla v MovieDetailService, tak by to byla asi extension method :)
po presunu do User jsem ju trosku upravil a ted uz neni jako extension method ale jako klasicka metoda, je to tak?

Reply:
jj, souhlasím.